src/world/Explosion.php

```diff
@@ -214,11 +248,34 @@ public function explodeB() : bool{
 				}
 				$this->world->setBlockAt($pos->x, $pos->y, $pos->z, $airBlock);
 			}
+
+			$fireBlock = VanillaBlocks::FIRE();
+
+			foreach($this->fireIgnitions as $ignition){
```

Comment:
I don’t like that you have to iterate over `fireIgnitions` on each `affectedBlocks` iteration. I think you could have `World::blockHash()` as a `fireIgnitions` element key instead of spl_object_id() (similar to how `affectedBlocks` are stored) and then check if `fireIgnitions` has that key on `affectedBlocks` iteration. If so and block below can support fire block, set fire instead of air